### PR TITLE
Marked Strings for Translation

### PR DESCRIPTION
**Pull request by clintonb**
This is a redux of #50 (which I messed up with git shenanigans). Feedback from @sarina has been incorporated. The basic setup has been moved to #58, which should be merged first.

@sarina @rocha @dsjen

### CODE REVIEW

File: analytics_dashboard/analytics_dashboard/settings/base.py [hunk 1]
```diff
@@ -349,3 +349,6 @@
 
 LOGIN_REDIRECT_URL = '/courses/'
 ########## END AUTHENTICATION
+
+# The display name of the platform to be used in templates/emails/etc.
+PLATFORM_NAME = 'edX'
```
Comment (sarina) [inline analytics_dashboard/analytics_dashboard/settings/base.py]: Would recommend you use a default placeholder for this (in edx-platform, "Your Platform Name Here"), then adjust it in your production environment however you are adjusting settings such as `SUPPORT_URL` (which this file just sets to example.com)

File: analytics_dashboard/analytics_dashboard/tests/test_templatetags.py [hunk 2]
```diff
@@ -15,3 +15,17 @@ def test_settings_value(self):
         with self.settings(FAKE_SETTING='edX'):
             # If setting is found, tag simply displays setting value.
             self.assertEqual(template.render(Context()), "edX")
+
+    def test_captureas(self):
```
Comment (sarina) [inline analytics_dashboard/analytics_dashboard/tests/test_templatetags.py:31]: Would recommend that you add a test to make sure this can handle Unicode, since many translations will include special Unicode characters.

Better to make two separate tests: one to test functionality; one to test Unicode, so it's easier to see where breakages happen.

File: analytics_dashboard/courses/templates/courses/enrollment_activity.html [hunk 3]
```diff
@@ -22,7 +23,9 @@
       <div class="section-content section-data-viz">
         <div id="enrollment-trend-view">
             <div class="line-chart-container"><div class="line-chart ">
-                {% include "loading.html" with message="Loading Trend..." %}
+                {% comment %}Translators: "Trend" is a graph of daily student enrollment{% endcomment %}
```
Comment (sarina) [inline analytics_dashboard/courses/templates/courses/enrollment_activity.html]: May also want to indicate if they should translate "Trend" to their language or not (eg "Please do not translate" or "Do translate into your language") - there are often strings that include variable names that they shouldn't translate adn it's hard to tell here whether or not that's the case.

File: analytics_dashboard/courses/templates/courses/enrollment_activity.html [hunk 4]
```diff
@@ -30,16 +33,17 @@
...
 
     <hr/>
 
     <div class="row">
       <div class="col-sm-12">
-            <span class="section-title-value small">Last updated
-                <span data-role="enrollment-last-updated">{{ summary.date|date }}</span>
+            <span class="section-title-value small">
+              {% captureas update_date %}<span data-role="enrollment-last-updated">{{ summary.date|date }}</span>{% endcaptureas %}
+              {% blocktrans %}Last updated: {{ update_date }}{% endblocktrans %}
```
Comment (sarina) [inline analytics_dashboard/courses/templates/courses/enrollment_activity.html:46]: Have you thought about how to i18n the date? (eg in many places a date is DD MM YY not MM DD YY like we do here)

File: analytics_dashboard/courses/templates/courses/enrollment_geography.html [hunk 5]
```diff
@@ -17,44 +17,55 @@
 
   <section class="view-section" data-section="enrollment-location">
     <h4 class="section-title">
-      <span class="section-title-value">Geographic Distribution</span>
-      <span class="section-title-note small">Where are my students learning from?</span>
+      <span class="section-title-value">{% blocktrans %}Geographic Distribution{% endblocktrans %}</span>
+      <span class="section-title-note small">{% blocktrans %}Where are my students learning
+        from?{% endblocktrans %}</span>
```
Comment (sarina) [inline analytics_dashboard/courses/templates/courses/enrollment_geography.html]: "Where are my students learning from?" seems extremely awkward to me. Just out of curiousity, have you had the doc team go through your product and comment on wording?

File: analytics_dashboard/courses/templates/courses/enrollment_geography.html [hunk 6]
```diff
@@ -17,44 +17,55 @@
...
     <div class="section-content section-data-viz">
-      <div class="world-map" data-view="world-map" data-title="Enrollment by Country" data-series-name="Enrollment">
+      <div class="world-map" data-view="world-map" data-title="{% trans "Enrollment by Country" %}"
+           data-series-name="{% trans "Enrollment" %}">
         {% comment %}
         The map is loaded via ajax, so display a loading message.  Everything inside of this div will be
         cleared when the map data loads.
         {% endcomment %}
-        {% include "loading.html" with message="Loading Map..." %}
+        {% trans "Loading Map..." as loading_msg %}
```
Comment (sarina) [inline analytics_dashboard/courses/templates/courses/enrollment_geography.html:42]: I want to see how this shows up in the .po file

File: analytics_dashboard/courses/templates/courses/enrollment_geography.html [hunk 7]
```diff
@@ -17,44 +17,55 @@
...
     <hr/>
 
     <h5 class="section-subtitle">
-      <span class="subsection-title-value"><strong>Country Enrollment Breakdown</strong></span>
-      <span class="subsection-title-note small">Which countries are represented by my students?</span>
+      <span class="subsection-title-value"><strong>
+        {% blocktrans %}Country Enrollment Breakdown{% endblocktrans %}
+      </strong></span>
+      <span class="subsection-title-note small">
+        {% blocktrans %}Which countries are represented by my students?{% endblocktrans %}
```
Comment (sarina) [inline analytics_dashboard/courses/templates/courses/enrollment_geography.html:54]: This is another sentence I find awkward

File: analytics_dashboard/courses/templates/courses/index.html [hunk 8]
```diff
@@ -24,7 +25,8 @@
 
     <div class="col-md-4">
       <div class="help-msg">
-        <h4>{% blocktrans %}New to edX Insights?{% endblocktrans %}</h4>
+        {% captureas platform_name %}{% settings_value 'PLATFORM_NAME' %}{% endcaptureas %}
+        <h4>{% blocktrans %}New to {{ platform_name }} Insights?{% endblocktrans %}</h4>
```
Comment (sarina) [inline analytics_dashboard/courses/templates/courses/index.html:29]: Should "Insights" be translated?

File: analytics_dashboard/templates/footer.html [hunk 9]
```diff
@@ -1,15 +1,24 @@
...
   <hr/>
   <div class="row">
     <div class="col-xs-6 col-md-6">
-      We want this dashboard to work for you! Please send suggestions, comments, and critiques to <a
-        class="feedback-email" href="mailto:{% settings_value 'FEEDBACK_EMAIL' %}?Subject=Feedback">{% settings_value 'FEEDBACK_EMAIL' %}</a>.
+      {% captureas email_link %}
+        <a class="feedback-email" href="mailto:{% settings_value 'FEEDBACK_EMAIL' %}?Subject=Feedback">
+        {% settings_value 'FEEDBACK_EMAIL' %}</a>{% endcaptureas %}
+      {% blocktrans %}
+        We want this dashboard to work for you! Please send suggestions, comments, and critiques to {{ email_link }}.
```
Comment (sarina) [inline analytics_dashboard/templates/footer.html:12]: :+1: this is A++ translation work, moving all the HTML out of the translated string!

**Comment by sarina** [comment]
Before we do a final merge I'd like to be sure that the translators comments you've added all correctly appear in the .po files.

File: analytics_dashboard/courses/templates/courses/enrollment_activity.html (hunk 4 above)
Comment (clintonb) [inline analytics_dashboard/courses/templates/courses/enrollment_activity.html:46]: This particular date would be rendered as `August 26, 2014`. We are using different formats in tables and on graphs. There is a ticket to update this, but I think it was pushed back. Yes, we have thought about it. No, we haven't yet implemented a solution.

File: analytics_dashboard/courses/templates/courses/enrollment_geography.html (hunk 5 above)
Comment (clintonb) [inline analytics_dashboard/courses/templates/courses/enrollment_geography.html]: Yes, but there have been a couple design changes. We need another review.

File: analytics_dashboard/courses/templates/courses/enrollment_geography.html (hunk 6 above)
Comment (clintonb) [inline analytics_dashboard/courses/templates/courses/enrollment_geography.html:42]:
``` po
#: courses/templates/courses/enrollment_geography.html:44
msgid "Loading Map..."
msgstr ""
```

File: analytics_dashboard/courses/templates/courses/index.html (hunk 8 above)
Comment (clintonb) [inline analytics_dashboard/courses/templates/courses/index.html:29]: I'm not sure. Seems like a branding question. Some brands translate, others do not. @shnayder any thoughts on this?

File: analytics_dashboard/courses/templates/courses/enrollment_activity.html (hunk 4 above)
Comment (sarina) [inline analytics_dashboard/courses/templates/courses/enrollment_activity.html:46]: OK. Would recommend putting a ticket in your backlog so if there are questions about it I can say it's in progress.

Comment (sarina) [inline analytics_dashboard/courses/templates/courses/enrollment_activity.html:46]: s/in progress/in the backlog/

File: analytics_dashboard/courses/templates/courses/enrollment_geography.html (hunk 6 above)
Comment (sarina) [inline analytics_dashboard/courses/templates/courses/enrollment_geography.html:42]: Cool! Just hadn't seen the translation syntax include the `as var_name` part before.

File: analytics_dashboard/courses/templates/courses/enrollment_activity.html (hunk 4 above)
Comment (clintonb) [inline analytics_dashboard/courses/templates/courses/enrollment_activity.html:46]: [AN-3015](https://openedx.atlassian.net/browse/AN-3015)

File: analytics_dashboard/courses/templates/courses/index.html (hunk 8 above)
Comment (clintonb) [inline analytics_dashboard/courses/templates/courses/index.html:29]: @sarina If you are alright with merging this PR, I will address the naming/branding issue in [AN-3158](https://openedx.atlassian.net/browse/AN-3158).

Comment (sarina) [inline analytics_dashboard/courses/templates/courses/index.html:29]: OK I had one other question on the PR, not in line, before I give thumbs

**Comment by sarina** [comment]
> Before we do a final merge I'd like to be sure that the translators comments you've added all correctly appear in the .po files.

**Comment by clintonb** [comment]
@sarina See https://gist.github.com/clintonb/cf6cf054df476559b743.

**Comment by sarina** [comment]
¡¡¡Vundarbar!!! :+1:

File: analytics_dashboard/courses/templates/courses/index.html (hunk 8 above)
Comment (shnayder) [inline analytics_dashboard/courses/templates/courses/index.html:29]: I know this PR is already merged, but... let's use "edX Insights" as the brand--don't translate either piece.

Comment (sarina) [inline analytics_dashboard/courses/templates/courses/index.html:29]: I would recommend using `{{ system_name }}` as the placeholder for the phrase "edX Insights", in that case.

Comment (clintonb) [inline analytics_dashboard/courses/templates/courses/index.html:29]: I was thinking the same thing, Sarina. We'll get to this shortly.

## 

Clinton Blackburn

Comment (rocha) [inline analytics_dashboard/courses/templates/courses/index.html:29]: Maybe `application_name` instead of `system_name`. The later can be a bit confusing.
